File: Python/FrogRiverOne.py
# https://app.codility.com/demo/results/trainingK9PKK5-364/
def solution(X, A):
    positions = set(e for e in A if e <= X)

    for P in range(1, X + 1):
        if P not in positions:
            return -1

    for K in range(len(A)):
        positions.discard(A[K])
        if len(positions) == 0:
            return K

    return -1

File: Java/FrogRiverOne.java
// https://app.codility.com/demo/results/trainingBRHYHJ-QDR/
import java.util.*;
import java.util.stream.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int solution(int X, int[] A) {
        int N = A.length;

        Set<Integer> positions = Arrays.stream(A)
            .filter(e -> e <= X)
            .boxed()
            .collect(Collectors.toSet());

        for (int P = 1; P <= X; P++) {
            if (!positions.contains(P)) {
                return -1;
            }
        }

        for (int K = 0; K < N; K++) {
            positions.remove(A[K]);
            if (positions.isEmpty()) {
                return K;
            }
        }

        return -1;
    }
}
